iceoryx_hoofs/test/moduletests/test_polymorphic_handler.cpp

```diff
@@ -0,0 +1,249 @@
+        Handler::reset();
+    }
+
+    void TearDown() override
+    {
+    }
+};
+
+TEST_F(PolymorphicHandler_test, handlerIsInitializedWithDefault)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "41bb4a5e-a916-4a6d-80c4-fed3a3d8d78b");
+    auto& handler = Handler::get();
+
+    EXPECT_EQ(handler.id(), DEFAULT_ID);
+}
+
+TEST_F(PolymorphicHandler_test, settingAlternateWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8b2f0cfe-f13c-4fa0-aa93-5ddd4f0904d1");
+    Handler::set(alternateGuard);
+    auto& handler = Handler::get();
+
+    EXPECT_EQ(handler.id(), ALTERNATE_ID);
+}
+
+TEST_F(PolymorphicHandler_test, alternatePointsToExternalMemory)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "85ce0e51-a1fe-490c-9012-7d539512ed38");
+    Handler::set(alternateGuard);
+
+    auto& handler = Handler::get();
+    auto* ptr = static_cast<Interface*>(&alternateHandler);
+
+    EXPECT_EQ(&handler, ptr);
+}
+
+TEST_F(PolymorphicHandler_test, explicitlySettingToDefaultWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "32e4d808-c848-4bf9-b878-e163ca825539");
+    Handler::set(alternateGuard);
+    Handler::set(defaultGuard);
+
+    auto& handler = Handler::get();
+    auto* ptr = static_cast<Interface*>(&defaultHandler);
+
+    EXPECT_EQ(&handler, ptr);
+}
+
+TEST_F(PolymorphicHandler_test, resetToDefaultWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ef8a99da-22a6-497e-b2ec-bf72cc3ae943");
+    Handler::set(alternateGuard);
+    Handler::reset();
+
+    auto& handler = Handler::get();
+    EXPECT_EQ(handler.id(), DEFAULT_ID);
+}
+
+TEST_F(PolymorphicHandler_test, settingAfterFinalizeCallsHook)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "171ac802-01b9-4e08-80a6-6f2defecaf6d");
+
+    auto& handler = Handler::get();
+    handler.reset();
+    alternateHandler.reset();
+
+    // note that all following tests will also call the after finalize
+    // hook but we only check if we care whether it was called
+    Handler::finalize();
+    Handler::set(alternateGuard);
+
+    // does the hook set the values to the corresponding arguments?
+    EXPECT_EQ(handler.value, DEFAULT_ID);
+    EXPECT_EQ(alternateHandler.value, ALTERNATE_ID);
+}
+
+TEST_F(PolymorphicHandler_test, resetAfterFinalizeCallsHook)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "996220e3-7985-4d57-bd3f-844987cf99dc");
+
+    auto& handler = Handler::get();
+    handler.reset();
+    alternateHandler.reset();
+
+    Handler::finalize();
+    Handler::reset();
+
+    EXPECT_EQ(handler.value, DEFAULT_ID);
+    EXPECT_EQ(alternateHandler.value, 0);
+}
+
+TEST_F(PolymorphicHandler_test, obtainingGuardWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "694f7399-598a-4918-b1e8-4b8546484245");
+    EXPECT_EQ(Guard<Handler>::count(), 1);
+    Guard<Handler> guard(Handler::guard());
+    EXPECT_EQ(Guard<Handler>::count(), 2);
+}
+
+class Activatable_test : public Test
+{
+  public:
+    void SetUp() override
+    {
+        internal::CaptureStderr();
```

Comment:
I'm not sure but I thought we want to get rid of this.

Reply:
Removed.